csharp/src/Ice/Incoming.cs

```diff
@@ -80,7 +80,11 @@ public bool reclaim()
             //_observer = null;
             Debug.Assert(_observer == null);
 
-            _os = null;
+            if(_os != null)
+            {
+                _os.reset();
```

Comment:
I don't understand why we do a reset in this reclaim. If it's correct, it deserves a comment.